src/components/organisms/PaymentModal/PaymentForm.tsx

```diff
@@ -100,26 +133,45 @@ const PaymentForm: React.FunctionComponent<PropsType> = ({
       {!clientSecret ? (
         <>Loading...</>
       ) : (
-        <form onSubmit={handleSubmit} className="payment-form-container">
+        <form className="payment-form-container">
           <input
             value={billingEmail ?? undefined}
             onChange={(event) => setBillingEmail(event.target.value)}
             className=""
           />
-          <CardSection />
-          <button
-            disabled={!stripe || isFormBeingSubmitted}
-            className="btn btn-primary btn-block confirm-order-button"
-            type="submit"
-          >
-            {!isFormBeingSubmitted ? (
-              "Confirm order"
-            ) : (
-              <div className="spinner-border" role="status">
-                <span className="sr-only">Loading...</span>
-              </div>
-            )}
-          </button>
+          <CardInput />
+          <div className="button-container">
+            <button
+              disabled={!stripe || isPaymentProceeding || isCardBeingSaved}
+              className="btn btn-primary btn-block submit-button"
+              type="submit"
+              onClick={(e) =>
+                handleSubmit(e, setIsCardBeingSaved, saveCardAndPay)
+              }
+            >
+              {!isCardBeingSaved ? (
+                "Save card and pay"
+              ) : (
+                <div className="spinner-border" role="status">
+                  <span className="sr-only">Loading...</span>
+                </div>
+              )}
+            </button>
+            <button
+              disabled={!stripe || isPaymentProceeding || isCardBeingSaved}
+              className="btn btn-primary btn-block submit-button"
+              type="submit"
+              onClick={(e) => handleSubmit(e, setIsPaymentProceeding, pay)}
```

Comment:
why doespay need to be passed here? It should be directly referenced in the `handleSubmit` fn